runtime/compiler/optimizer/FearPointAnalysis.cpp

```diff
@@ -34,10 +34,20 @@
 #include "infra/BitVector.hpp"
 #include "infra/Checklist.hpp"
 
-bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp)
+
+bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp, TR::Node *virtualGuardNode)
    {
-   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear") == NULL);
-   return kill && !(!comp->getOption(TR_DisableVectorAPIExpansion) && comp->getMethodSymbol()->hasVectorAPI());
+   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear")) == NULL;
+   if (kill && !comp->getOption(TR_DisableVectorAPIExpansion))
+      {
+      TR_VirtualGuard *guardInfo = comp->findVirtualGuardInfo(virtualGuardNode);
+      if (guardInfo != NULL && guardInfo->isInlineGuard())
```

Comment:
FYI, since eclipse/omr#6712, `findVirtualGuardInfo()` always returns non-null; if the result would be null, we'll fail an assertion instead

Reply:
removed the `guardInfo != NULL` query